gapic-generator/gapic-generator.gemspec

```diff
@@ -38,12 +38,15 @@ Gem::Specification.new do |spec|
   spec.executables   = ["gapic-generator", "protoc-gen-ruby_gapic"]
   spec.require_paths = ["lib"]
 
-  spec.required_ruby_version = ">= 2.7"
+  spec.required_ruby_version = ">= 3.1"
 
   spec.add_dependency "actionpack", "~> 5.2"
+  spec.add_dependency "base64", "~> 0.2.0" # needed for actionpack
```

Comment:
I think we don't need this (or mutex_m or ostruct) if we stay on Ruby 3.1 instead of jumping to 3.4.

Reply:
ack